savings-account/src/math.rs

```diff
@@ -33,8 +32,7 @@ pub trait MathModule {
         borrowed_amount: &BigUint,
         total_pool_reserves: &BigUint,
     ) -> BigUint {
-        let bp = BigUint::from(BASE_PRECISION);
-        &(borrowed_amount * &bp) / total_pool_reserves
+        &(borrowed_amount * BASE_PRECISION) / total_pool_reserves
```

Comment:
It's possible that the reference is superfluous, but not sure. Looks good anyway.

Reply:
You're right, it's no needed. Removed.